grants/milestone_deliveries/season_2/package_manager/milestone1-part1.md

```diff
@@ -0,0 +1,39 @@
+# Milestone Delivery :mailbox: - Milestone 1, Part 1
+
+**The [invoice form :pencil:](TODO) has been filled out correctly for this milestone and the delivery is according to the official [milestone delivery guidelines](https://github.com/radicle-dev/radicle-grants/blob/main/docs/deliverables_guideline.md).**  
+
+* **Application Document:** [Grant Application - Package Manager](https://radicle.community/t/grant-application-package-manager/3122)
+* **Milestone Number:** Milestone 1, part 1
+
+**Context**
+
+The scope of this first part of the first milestone was to *Generate an npm-compatible package from source code obtained from Radicle*. While issues were discovered during this research, a solution for moving forward was found.
+
+**Deliverables**
+
+The [research notes](./milestone1-part1-details.md) are presented here as a deliverable for the first part of the first milestone. A summary video is included, in addition to written notes. It was determined that it **is** possible to build a JavaScript application without any dependence on GitHub or npm. JavaScript packages (dependencies) can be retrieved directly from Radicle Seed nodes (as git repositories).
+
+| Number | Deliverable | Link | Notes |
+| ------------- | ------------- | ------------- |------------- |
+| 1. | Generate an npm-compatible package from source code obtained from Radicle | [Details](./milestone1-part1-details.md) | Research notes |
+
+**Additional Information**
+
+Additional guidance is needed before moving on to the second and third parts of the fist milestone, but it is possible to move on directly to the fourth part. Here are the four parts of the first milestone:
+
+| Number | Deliverable              | Specification                                                |
+| ------ | ------------------------ | ------------------------------------------------------------ |
+| 1.     | Package code             | Generate an npm-compatible package from source code obtained from Radicle |
+| 2.     | Package signing          | Sign packages with Radicle keys                                           |
+| 3.     | Package upload           | Research best way to upload packages                                      |
+| 4.     | Verdaccio integration    | Research the best way to integrate with Verdaccio                         |
+
+There were two outcomes from the first part of the research:
+1. It **is not** possible to retrieve a binary packages directly from a Seed node.
```

Comment:
My first thought is that it would be great to support fetching raw binaries. My second thought was then, wait, what are the security considerations there? Could we do better by knowing this is a verified `tar.gz` somehow? Would definitely benefit from some considerations of what the implications are for serving raw data and how that reflects on Radicle's ethos :)

Reply:
If Radicle *did* make it possible to serve specific files from a repository, it would feature match GitHub. GitHub already does this.

I'm having a hard time thinking of any security issues that would exist around that feature.

Reply:
Let us not forget that GitHub has the advantage of being centralised :) For someone to trust a GitHub repository tends to mean that they trust that organisation or individual, generally through social elements -- e.g. I know `facebook/react` is the go-to implementation. 

For Radicle, I think this can be true as well if you only use seed nodes that you trust, along with Radicle IDs that you trust. But if this ends up being something more like, "get this project artifact from any seed node using `rad:git:abc/v1.tar.gz`, well then I would think you'd want ways of verifying that artifact, based on the trust system of the project identity history and delegates.

In general, I feel like it's a healthy mindset to be security conscious when arbitrary bytes and executables are involved -- not that I'm always wearing a tin-foil hat or anything :sweat_smile: